lib/gemstash/http_client.rb

```diff
@@ -59,8 +51,28 @@ def get(path)
       end
     end
 
+    def head(path)
+      response = request(:head, path)
+      if block_given?
+        yield response.headers
+      else
+        response.headers
+      end
+    end
+
   private
 
+    def request(method, path)
+      response = with_retries do
+        @client.public_send(method, path) do |request|
+          request.headers["User-Agent"] = @user_agent
+          request.options.open_timeout = 2
+        end
+      end
+      raise Gemstash::WebError.new(response.body, response.status) unless response.success?
```

Comment:
The head is so gemstash does not push the body to this client as it is not necessary, but gemstash will make a full get to the upstream, so it will cache the gem. But will not write the body back to this client.